packages/kbn-management/settings/application/hooks/use_fields.ts

```diff
@@ -28,8 +28,22 @@ export const useFields = (scope: UiSettingsScope, query?: Query): FieldDefinitio
     isCustom: (key) => isCustomSetting(key, scope),
     isOverridden: (key) => isOverriddenSetting(key, scope),
   });
-  if (query) {
-    return Query.execute(query, fields);
+  if (query && fields.length) {
+    let ast = Ast.create([]);
+
+    query.ast.clauses.forEach((clause) => {
+      if (clause.type !== 'field' || Object.keys(fields[0]).includes(clause.field)) {
```

Comment:
I don't think we should cover `field` type queries for the `FieldDefinition` properties as these properties are not exposed and visible to the user. Also, the `Object.keys(fields[0]).includes(clause.field)` check is not complete as the first field may not contain some of the optional properties (e.g. `deprecation`).

I would suggest to just check for field-type clauses:
```suggestion
      if (clause.type === 'field') {
        // If the clause value contains `:`, add it as a term clause
        // This allows searching for settings that include `:` in their names
        ast = ast.addClause({
          type: 'term',
          match: 'must',
          value: `${clause.field}:${clause.value}`,
        });
      } else {
        ast = ast.addClause(clause);
      }
```

Reply:
@ElenaStoeva  Got it. And maybe also needs to determine whether `clause.field` is not included in the filters. 

Otherwise, `categories` filter will not working.

Do you think it's just check `clause.field !== `categories``, or pass in filters then check filters field?

```suggestion
    let ast = Ast.create([]);
    const filterFields = filters
      ?.filter((item) => item.type !== 'custom_component')
      ?.map((item) => (item as Exclude<SearchFilterConfig, CustomComponentFilterConfigType>).field);

    query.ast.clauses.forEach((clause) => {
      if (clause.type === 'field' && !filterFields.includes(clause.field)) {
        // If the clause value contains `:`, and not in field filters, add it as a term clause
        // This allows searching for settings that include `:` in their names
        ast = ast.addClause({
          type: 'term',
          match: 'must',
          value: `${clause.field}:${clause.value}`,
        });
      } else {
        ast = ast.addClause(clause);
      }
    });
```

Reply:
Good call about the categories filter! I think it's okay to just check if `clause.field !== 'categories'` as I don't think there are plans to add more filters in the future, otherwise only a small change here would be needed.

We could also simplify this logic some more:
```
  if (query && fields.length) {
    const clauses: Clause[] = query.ast.clauses.map((clause) =>
      // If the clause value contains `:` and is not a category filter, add it as a term clause
      // This allows searching for settings that include `:` in their names
      clause.type === 'field' && clause.field !== 'categories'
        ? {
            type: 'term',
            match: 'must',
            value: `${clause.field}:${clause.value}`,
          }
        : clause
    );

    return Query.execute(new Query(Ast.create(clauses), undefined, query.text), fields);
  }
```

Reply:
fixed, Thanks for your great suggestion. 🎉